### DISCUSSION TITLE
为什么微调的时候，准备的数据格式上要求如果带有tools，就不能带有system的content？

### DISCUSSION THREAD

**Discussion by linboyang** [THUDM/GLM-4#153] [Q&A] [answered]
![image](https://github.com/THUDM/GLM-4/assets/22309864/266d8b20-1fb0-47d5-a36a-e4c04d5f30db)
小白问，求大佬解释一下原因

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
因为GLM-4模型带工具调用的时候必须用我们已经写好的system，在tokenizer实现中你会看到是如何构建的
